src/components/Navbar.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState, useRef } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import { useKanban } from '../hooks/useKanban';
+
+export function Navbar() {
+  const { user, signInWithGoogle, signOut } = useAuth();
+  const { isDark, toggleTheme } = useTheme();
+  const { boardTitle, updateBoardTitle, addTask, updateColumn, columns } = useKanban();
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(boardTitle);
+  const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [showForms, setShowForms] = useState(false);
+  const titleInputRef = useRef(null);
+
+  const handleTitleClick = () => {
+    if (user) {
+      setIsEditingTitle(true);
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleChange = (e) => {
+const handleTitleBlur = async () => {
+  setIsEditingTitle(false);
+  if (editedTitle.trim() && editedTitle !== boardTitle) {
+    try {
+      await updateBoardTitle(editedTitle.trim());
+    } catch (error) {
+      setEditedTitle(boardTitle);
+      console.error('Failed to update board title:', error);
+      // Add user feedback mechanism here
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
console.error('Failed to update board title:', error);
showErrorNotification('Failed to update board title. Please try again.');
```
The error handling could be enhanced by implementing a proper error notification system instead of just logging to console. Consider using a toast notification or similar UI feedback mechanism.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;